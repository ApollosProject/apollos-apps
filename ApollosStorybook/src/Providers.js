import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { LiveProvider } from '@apollosproject/ui-connected';
import ClientProvider, { client } from './client';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider>
    <MediaPlayerProvider>
        <LiveProvider>
          <Providers
            themeInput={customTheme}
            iconInput={customIcons}
            {...props}
          />
        </LiveProvider>
    </MediaPlayerProvider>
  </ClientProvider>
);

export default AppProviders;
