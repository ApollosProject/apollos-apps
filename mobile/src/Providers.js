import { nest } from 'recompose';

import ClientProvider from '/mobile/client';
import { ThemeProvider } from '/mobile/ui/theme';
import { WebBrowserProvider } from '/mobile/ui/WebBrowser';

export default nest(ClientProvider, ThemeProvider, WebBrowserProvider);
