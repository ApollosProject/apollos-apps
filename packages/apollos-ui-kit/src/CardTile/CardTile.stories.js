import React from 'react';
import { View, Text } from 'react-native';
import { storiesOf } from '@apollosproject/ui-storybook';

import CardTile from '.';

const containerStyles = {
  height: 300,
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: '#f7f7f7',
};

storiesOf('ui-kit/CardTile', module)
  .add('Default', () => (
    <View style={containerStyles}>
      <CardTile title={'Why Jesus is Timeless'} />
    </View>
  ))
  .add('Skeleton', () => (
    <View style={containerStyles}>
      <CardTile
        number={7}
        title={'Why Jesus is Timeless'}
        byLine={'Marty McFly'}
        date={'Sat Oct 26 1985 01:24:00 GMT+0008 (UTC)'}
        isLoading
      />
    </View>
  ))
  .add('With Number Box', () => (
    <View style={containerStyles}>
      <CardTile number={7} title={'Why Jesus is Timeless'} />
    </View>
  ))
  .add('With byLine', () => (
    <View style={containerStyles}>
      <CardTile
        number={7}
        title={'Why Jesus is Timeless'}
        byLine={'Marty McFly'}
      />
    </View>
  ))
  .add('With date', () => (
    <View style={containerStyles}>
      <CardTile
        number={7}
        title={'Why Jesus is Timeless'}
        byLine={'Marty McFly'}
        date={'Sat Oct 26 1985 01:24:00 GMT+0008 (UTC)'} // this snapshot will expire in a year
      />
    </View>
  ))
  .add('With Children', () => (
    <View style={containerStyles}>
      <CardTile>
        <Text>Biff Tannen was here</Text>
      </CardTile>
    </View>
  ));
