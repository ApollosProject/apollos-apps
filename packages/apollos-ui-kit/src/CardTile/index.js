import React from 'react';
import { Dimensions, View, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { compose, pure } from 'recompose';
import moment from 'moment';

import Placeholder from '../Placeholder';
import { withIsLoading } from '../isLoading';
import { withTheme, withThemeMixin } from '../theme';
import styled from '../styled';
import { H3, H5, H6 } from '../typography';
import { CardContent, CardActions } from '../Card';
import ChannelLabel from '../ChannelLabel';

const enhance = compose(
  withIsLoading,
  withThemeMixin(({ theme }) => ({
    type: 'light',
    colors: {
      background: {
        inactive: theme.colors.lightSecondary,
      },
    },
  })),
  withTheme(),
  pure
);

const TileSpacer = styled(({ theme }) => ({
  padding: theme.sizing.baseUnit / 2,
}))(View);

const Tile = styled(({ theme }) => ({
  width: Dimensions.get('window').width * 0.66,
  aspectRatio: 1,
  borderRadius: theme.sizing.baseUnit,
  backgroundColor: theme.colors.lightTertiary,
}))(View);

const PlaceholderOverflowFix = styled(({ theme }) => ({
  ...StyleSheet.absoluteFillObject,
  justifyContent: 'center',
  borderRadius: theme.sizing.baseUnit,
  overflow: 'hidden',
}))(View);

const TileNumber = styled(({ theme, size }) => ({
  position: 'absolute',
  top: 0,
  left: 0,
  width: theme.helpers.rem(1 * (size < 2 ? 2 : size)),
  height: theme.helpers.rem(1 * (size < 2 ? 2 : size)),
  justifyContent: 'center',
  alignItems: 'center',
  borderTopLeftRadius: theme.sizing.baseUnit,
  borderBottomRightRadius: theme.sizing.baseUnit,
  backgroundColor: theme.colors.white,
}))(View);

const relativeTime = (date) => {
  if (moment.updateLocale) {
    moment.updateLocale('en', {
      relativeTime: {
        future: 'in %s',
        past: '%s ago',
        s: 's',
        m: 'm',
        mm: '%dm',
        h: '1h',
        hh: '%dh',
        d: '1d',
        dd: '%dd',
        M: '1mo',
        MM: '%dmo',
        y: '1y',
        yy: '%dy',
      },
    });
  }

  const time = moment(new Date(date));
  return time.fromNow(true); // true omits 'ago'
};

const CardTile = enhance(
  ({
    number,
    title,
    byLine,
    date,
    style: styleProp = {},
    isLoading,
    theme,
    children,
    ...otherProps
  }) => (
    <TileSpacer collapsable={false}>
      <Tile style={styleProp} {...otherProps}>
        <PlaceholderOverflowFix>
          {typeof number === 'undefined' ? null : (
            <TileNumber size={number.toString().length}>
              <Placeholder.Media
                size={theme.helpers.rem(
                  1.25 *
                    (number.toString().length < 2
                      ? 2
                      : number.toString().length)
                )}
                onReady={!isLoading}
              >
                <View>
                  <H5>{number}</H5>
                </View>
              </Placeholder.Media>
            </TileNumber>
          )}

          {typeof title === 'undefined' ? null : (
            <CardContent>
              <H3>{title}</H3>
            </CardContent>
          )}

          {byLine ? (
            <CardActions>
              <ChannelLabel
                label={byLine}
                icon={'video'}
                isLoading={isLoading}
                withFlex
              />
              {typeof date === 'undefined' ? null : (
                <H6>{relativeTime(date)}</H6>
              )}
            </CardActions>
          ) : null}

          {children}
        </PlaceholderOverflowFix>
      </Tile>
    </TileSpacer>
  )
);

CardTile.propTypes = {
  title: PropTypes.string,
  number: PropTypes.number,
  byLine: PropTypes.string,
  date: PropTypes.string,
  style: PropTypes.any, // eslint-disable-line
  isLoading: PropTypes.bool,
};

export default CardTile;
