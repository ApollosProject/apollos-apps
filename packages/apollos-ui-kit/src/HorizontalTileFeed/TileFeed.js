import { FlatList } from 'react-native';

import { withTheme } from '../theme';

const TileFeed = withTheme(({ theme }) => ({
  contentContainerStyle: {
    // The horizontal padding keeps cards on grid (the other half of the baseUnit lives on the
    // card). The vertical padding is purly stylistic.
    padding: theme.sizing.baseUnit / 2,
    paddingBottom: 0,
  },
}))(FlatList);

export default TileFeed;
