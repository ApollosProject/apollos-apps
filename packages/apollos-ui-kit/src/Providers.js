import { nest } from 'recompose';

import { ThemeProvider } from './theme';
import { LayoutProvider } from './LayoutContext';

export default nest(ThemeProvider, LayoutProvider);
