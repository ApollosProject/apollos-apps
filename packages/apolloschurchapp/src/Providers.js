import { nest } from 'recompose';

import ClientProvider from 'apolloschurchapp/src/client';
import { ThemeProvider } from 'apolloschurchapp/src/ui/theme';
import { WebBrowserProvider } from 'apolloschurchapp/src/ui/WebBrowser';

export default nest(ClientProvider, ThemeProvider, WebBrowserProvider);
