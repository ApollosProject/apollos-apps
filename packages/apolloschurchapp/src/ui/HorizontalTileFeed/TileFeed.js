import { FlatList } from 'react-native';

import styled from 'apolloschurchapp/src/ui/styled';

const TileFeed = styled(({ theme, tileHeight }) => ({
  height: tileHeight,
  paddingBottom: theme.sizing.baseUnit / 2,
}))(FlatList);

export default TileFeed;
