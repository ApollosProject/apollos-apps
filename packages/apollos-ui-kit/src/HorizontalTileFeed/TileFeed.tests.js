import React from 'react';
import renderer from 'react-test-renderer';

import Providers from '../Providers';
import CardTile from '../CardTile';

import TileFeed from '.';

describe('The TileFeed component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Providers>
        <TileFeed
          content={[
            {
              node: {
                id: 'fakeId0',
                title: 'Why Jesus is Timeless',
              },
            },
            {
              node: {
                id: 'fakeId1',
                title: 'Why Jesus is Timeless',
              },
            },
          ]}
          loadingStateObject={{
            id: 'fakeId0',
            title: '',
            isLoading: false,
          }}
          renderItem={({ item, index }) => (
            <CardTile
              number={index + 1}
              title={item.title}
              isLoading={item.isLoading}
            />
          )}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('renders empty state', () => {
    const tree = renderer.create(
      <Providers>
        <TileFeed
          refreshing
          content={[]}
          loadingStateObject={{
            id: 'fakeId0',
            title: '',
            isLoading: true,
          }}
          renderItem={({ item, index }) => (
            <CardTile
              number={index + 1}
              title={item.title}
              isLoading={item.isLoading}
            />
          )}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
