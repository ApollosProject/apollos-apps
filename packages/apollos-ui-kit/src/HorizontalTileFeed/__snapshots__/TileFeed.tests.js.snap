// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The TileFeed component renders correctly 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "padding": 8,
          "paddingBottom": 0,
        }
      }
      data={
        Array [
          Object {
            "node": Object {
              "id": "fakeId0",
              "title": "Why Jesus is Timeless",
            },
          },
          Object {
            "node": Object {
              "id": "fakeId1",
              "title": "Why Jesus is Timeless",
            },
          },
        ]
      }
      decelerationRate="fast"
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      initialScrollIndex={0}
      keyExtractor={[Function]}
      loadingStateObject={
        Object {
          "id": "fakeId0",
          "isLoading": false,
          "title": "",
        }
      }
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshing={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      snapToAlignment="start"
      snapToInterval={512}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "padding": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "aspectRatio": 1,
                  "backgroundColor": "#A5A5A5",
                  "borderRadius": 16,
                  "width": 495,
                }
              }
            >
              <View
                style={
                  Object {
                    "borderRadius": 16,
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  size={1}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#FFFFFF",
                      "borderBottomRightRadius": 16,
                      "borderTopLeftRadius": 16,
                      "height": 32,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "top": 0,
                      "width": 32,
                    }
                  }
                >
                  <View>
                    <Text
                      style={
                        Object {
                          "color": "#303030",
                          "fontFamily": "InterUI-Medium",
                          "fontSize": 14,
                          "lineHeight": 20.16,
                        }
                      }
                    >
                      1
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "padding": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "aspectRatio": 1,
                  "backgroundColor": "#A5A5A5",
                  "borderRadius": 16,
                  "width": 495,
                }
              }
            >
              <View
                style={
                  Object {
                    "borderRadius": 16,
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  size={1}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#FFFFFF",
                      "borderBottomRightRadius": 16,
                      "borderTopLeftRadius": 16,
                      "height": 32,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "top": 0,
                      "width": 32,
                    }
                  }
                >
                  <View>
                    <Text
                      style={
                        Object {
                          "color": "#303030",
                          "fontFamily": "InterUI-Medium",
                          "fontSize": 14,
                          "lineHeight": 20.16,
                        }
                      }
                    >
                      2
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`The TileFeed component renders empty state 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "padding": 8,
          "paddingBottom": 0,
        }
      }
      data={Array []}
      decelerationRate="fast"
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      initialScrollIndex={0}
      keyExtractor={[Function]}
      loadingStateObject={
        Object {
          "id": "fakeId0",
          "isLoading": true,
          "title": "",
        }
      }
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshing={true}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      snapToAlignment="start"
      snapToInterval={512}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
